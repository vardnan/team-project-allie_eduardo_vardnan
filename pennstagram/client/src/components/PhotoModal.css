
.photoModalWrapper {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-content: flex-start;
    align-items: stretch;
    z-index: 10;
    position: fixed;
    width: 850px;
    height: 600px;
    overflow: hidden;
    background-color: white;
    padding: 0;
    margin: 0;
    margin-top: 50px;
    border: 1px solid #dcdcdc;
    border-radius: 4px;
}

.photoModalImage {
    order: 0;
    width: 100%;
    height: 100%;
}

.photoModalImage img {
    width: 100%;
    height: 100%;
}

.photoModalSidePanel {
    display: flex;
    flex-direction: column;
    gap: 5px;
    height: 100%;
    order: 1;
    position: relative;
    width: 35%;
    margin: 0;
}

.closeButton {
    position: absolute;
    font-size: 10pt;
    font-weight:500;
    height: 25px;
    width: 25px;
    top: 0;
    right: 0;
    padding: 0;
    margin: 0;
    border: 0;
    border-bottom-left-radius: 4px;
    transition: 0.1s ease-in-out;
    opacity: 0.9;
}

.closeButton:hover {
    background-color: #ff5571;
    color: white;
    cursor: pointer;
}

.sidePanelHeader {
    order: 0;
    width: 100%;
    display: flex;
    flex-direction: row;
    padding: 10px;
    align-items: center;
    margin: 0;
    margin-bottom: 0px;
    border-bottom: 1px solid #dcdcdc;
}

.sidePanelComments {
    overflow-y: auto;
    order: 1;
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
}

.sidePanelHeader img {
    order: 0;
    border-radius: 100%;
    width: 40px;
    height: 40px;
    padding: 0;
    margin: 0;
    margin-right: 10px;
}

.sidePanelHeader h3 {
    order: 1;
    margin: 0;
    width: 100%;
    border-bottom: 1px;
    border-color: black;
    font-size: 11pt;
}

.actionSection {
    order: 2;
    width: 100%;
    display: flex;
    flex-direction: column;
    order: 2;
    background-color: white;
    bottom: 0;
    left: 0;
    right: 0;
    border-top: 1px solid #dcdcdc;
    margin: 0;
    padding: 0;
}

.actionButtons {
    display: flex;
    flex-direction: row;
    padding: 5px;
    order: 0;
}

.actionButtons button {
    margin: 3px;
}

.commentInput {
    padding: 10px;
    border-top: 1px solid #dcdcdc;
    order: 1;
}

.commentInput textarea {
    border: none;
    resize: none;
    overflow: auto;
    width: 100%;
    height: 100%;
    padding: 3px;
    margin: 0px;
}

.commentInput textarea:focus {
    outline: none;
    box-shadow: none;
}

.actionButtonContainer {
    display: flex;
    flex-direction: column;
}

.actionButtonContainer p {
    color: rgb(56, 56, 56);
    font-size: 11pt;
    font-weight: bold;
    padding-top: 5px;
    padding-bottom: 11px;
    padding-left: 11px;
    margin: 0;
}