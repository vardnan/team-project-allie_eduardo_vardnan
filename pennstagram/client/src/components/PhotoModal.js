import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import './PhotoModal.css';
import { motion, AnimatePresence } from 'framer-motion';
import io from 'socket.io-client';
import CommentSection from './CommentSection';
import CommentInput from './CommentInput';

function PhotoModal({
  _id,
  isVideo,
  imageUrl,
  caption,
  username,
  profileImage,
  onClose,
  comments: initialComments,
  currentUser,
  likes: initialLikes,
  // added this
  // postId,
}) {
  const [comments, setComments] = useState(initialComments);
  const [likes] = useState(initialLikes);

  // console.log('Received postId:', postId);

  const fadeInOut = {
    hidden: { opacity: 0, scale: 0.95 },
    visible: { opacity: 1, scale: 1 },
  };

  useEffect(() => {
    // const socket = io('http://localhost:8080');
    const socket = io('');
    // Listen for new comment events
    socket.on('newComment', (newComment) => {
      // eslint-disable-next-line
      // console.log('new comment received');
      // Update the comments state with the new comment
      setComments((prevComments) => [...prevComments, newComment]);
    });

    return () => {
      // Disconnect the socket when the component unmounts
      socket.disconnect();
    };
  }, []);

  return (
    <AnimatePresence>
      <motion.div
        className="photoModalWrapper"
        initial="hidden"
        animate="visible"
        exit="hidden"
        variants={fadeInOut}
        transition={{ duration: 0.3, ease: [0.04, 0.62, 0.23, 0.98] }}
      >
        <div className="photoModalImage">
        {isVideo ? (
          <video controls width="100%" height="100%">
            <source src={imageUrl} type="video/mp4" />
            Your browser does not support the video tag.
          </video>
        ) : (
          <img src={imageUrl} alt={caption} />
        )}
        </div>
        <div className="photoModalSidePanel">
          <button className="closeButton" type="button" onClick={onClose}>
            X
          </button>
          <div className="sidePanelHeader">
            <img src={profileImage} alt={username} className="profileImg" />
            <h3>{username}</h3>
          </div>
          <div className="sidePanelComments">
            <CommentSection
              comments={comments}
              profileImage={profileImage}
              caption={caption}
              postUser={username}
            />
          </div>
          <div className="actionSection">
            <div className="actionButtonContainer">
              <div className="actionButtons">
              </div>
              <p>
                {likes.length}
                {' '}
                likes
              </p>
            </div>
            <div className="commentInput">
              <CommentInput
                postId = {_id.toString()}
                currentUser={currentUser}
                comments={comments}
                setComments={setComments}
              />
            </div>
          </div>
        </div>
      </motion.div>
    </AnimatePresence>
  );
}

PhotoModal.propTypes = {
  isVideo: PropTypes.bool.isRequired,
  username: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  profileImage: PropTypes.string.isRequired,
  caption: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  comments: PropTypes.array.isRequired,
  currentUser: PropTypes.object.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  likes: PropTypes.array.isRequired,
  _id: PropTypes.string.isRequired,
};

export default PhotoModal;
