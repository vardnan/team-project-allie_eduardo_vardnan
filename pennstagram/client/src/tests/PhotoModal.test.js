import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PhotoModal from '../components/PhotoModal';

jest.mock('../components/LikeButton', () => {
  return function MockedLikeButton() {
    return <button data-testid="like-button"></button>;
  };
});

jest.mock('../components/CommentSection', () => {
  return function MockedCommentSection() {
    return <div data-testid="comment-section"></div>;
  };
});

jest.mock('../components/CommentButton', () => {
  return function MockedCommentButton() {
    return <button data-testid="comment-button"></button>;
  };
});

jest.mock('../components/CommentInput', () => {
  return function MockedCommentInput() {
    return <textarea data-testid="comment-input"></textarea>;
  };
});

// Testing rendering
test('Testing that the PhotoModal is rendering correctly', () => {
  const props = {
    username: 'Alice',
    image: 'image.jpg',
    profileImage: 'profile.jpg',
    caption: 'Caption text',
    onClose: jest.fn(),
    comments: [],
    currentUser: 'Bob',
    likes: [],
  };

  const { getByAltText } = render(<PhotoModal {...props} />);
  expect(getByAltText('Caption text')).toBeInTheDocument();
});

// Snapshot testing
test('Testing that the PhotoModal matches snapshot', () => {
  const props = {
    username: 'Alice',
    image: 'image.jpg',
    profileImage: 'profile.jpg',
    caption: 'Caption text',
    onClose: jest.fn(),
    comments: [],
    currentUser: 'Bob',
    likes: [],
  };

  const component = renderer.create(<PhotoModal {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// Testing events

test('Testing the onClose function when the close button is clicked', () => {
  const mockOnClose = jest.fn();
  const props = {
    username: 'Alice',
    image: 'image.jpg',
    profileImage: 'profile.jpg',
    caption: 'Caption text',
    onClose: mockOnClose,
    comments: [],
    currentUser: 'Bob',
    likes: [],
  };

  const { container } = render(<PhotoModal {...props} />);
  const closeButton = container.querySelector('.closeButton');
  fireEvent.click(closeButton);
  expect(mockOnClose).toHaveBeenCalledTimes(1);
});
