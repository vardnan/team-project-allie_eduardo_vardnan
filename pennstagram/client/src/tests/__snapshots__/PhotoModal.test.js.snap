// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing that the PhotoModal matches snapshot 1`] = `
<div
  className="photoModalWrapper"
  style={
    Object {
      "opacity": 0,
      "transform": "scale(0.95) translateZ(0)",
    }
  }
>
  <div
    className="photoModalImage"
  >
    <img
      alt="Caption text"
      src="image.jpg"
    />
  </div>
  <div
    className="photoModalSidePanel"
  >
    <button
      className="closeButton"
      onClick={[MockFunction]}
    >
      X
    </button>
    <div
      className="sidePanelHeader"
    >
      <img
        alt="Alice"
        className="profileImg"
        src="profile.jpg"
      />
      <h3>
        Alice
      </h3>
    </div>
    <div
      className="sidePanelComments"
    >
      <div
        data-testid="comment-section"
      />
    </div>
    <div
      className="actionSection"
    >
      <div
        className="actionButtonContainer"
      >
        <div
          className="actionButtons"
        >
          <button
            data-testid="like-button"
          />
          <button
            data-testid="comment-button"
          />
        </div>
        <p>
          0
           likes
        </p>
      </div>
      <div
        className="commentInput"
      >
        <textarea
          data-testid="comment-input"
        />
      </div>
    </div>
  </div>
</div>
`;
